dart/lib/src/sentry.dart

```diff
@@ -53,23 +54,10 @@ class Sentry {
 
   static Future<void> _initDefaultValues(
       SentryOptions options, AppRunner? appRunner) async {
-    // We infer the enviroment based on the release/non-release and profile
-    // constants.
-    var environment = options.platformChecker.isReleaseMode()
-        ? defaultEnvironment
-        : options.platformChecker.isProfileMode()
-            ? 'profile'
-            : 'debug';
-
-    // if the SENTRY_ENVIRONMENT is set, we read from it.
-    options.environment = const bool.hasEnvironment('SENTRY_ENVIRONMENT')
-        ? const String.fromEnvironment('SENTRY_ENVIRONMENT')
-        : environment;
-
-    // if the SENTRY_DSN is set, we read from it.
-    options.dsn = const bool.hasEnvironment('SENTRY_DSN')
-        ? const String.fromEnvironment('SENTRY_DSN')
-        : options.dsn;
+    var environment = options.platformChecker.environment;
+    options.environment = options.environment ?? environment;
+
+    setEnvironmentVariables(options, EnvironmentVariables());
```

Comment:
yo uset `options.environment` outside of the method `setEnvironmentVariables`, ideally, we'd have a single method that does everything, pass `environment` as a param if its the case

Reply:
Makes sense